src/ClientModel/include/ClientModel/CaptureData.h

```diff
@@ -258,6 +267,9 @@ class CaptureData {
       thread_state_slices_;  // For each thread, assume sorted by timestamp and not overlapping.
   mutable std::unique_ptr<absl::Mutex> thread_state_slices_mutex_ = std::make_unique<absl::Mutex>();
 
+  // Only access missing_data_intervals_ from the main thread.
```

Comment:
> I guess this was renamed at some point and is actually supposed to be: incomlete_data_intervals_

Eh yeah, I didn't manage to rename it without leaving traces... Done.

Reply:
> is OnMetadataEvent called from the main thread?

Indeed it's not. But for simplicity everything it does is on the `main_thread_executor_` because these events are rare.